File: app/lib/models/service.dart
class Service {
  String containerID;
  String name;
  List<Map<String, String>> ports = [];
  String status;
  String createdAt;
  String? image;
  List<Map<String, String>> environments = [];
  List<Map<String, String>> volumes = [];

  Service(this.containerID, this.name, this.ports, this.status, this.createdAt);

  Service.fromJson(Map<String, dynamic> json)
      : containerID = json['container_id'],
        name = json['name'],
        ports = json['ports'],
        status = json['status'],
        createdAt = json['created_at'];

  Service.newService()
      : containerID = '',
        name = '',
        ports = [],
        status = '',
        createdAt = '';

  String get parsePort {
    String result = "";

    for (Map<String, String> port in ports) {
      result = result + "${port['host']} -> ${port['container']}\n";
    }

    return result.trim();
  }
}

class ServiceState {
  final List<Service> services;
  final bool loading;

  ServiceState(this.services, {this.loading = false});

  ServiceState.from(ServiceState state,
      {List<Service>? newServices, bool newLoading = false})
      : services = newServices ?? state.services,
        loading = newLoading;
}

File: server/source/models/service.py
class DockerService():
    """Class that abstracts docker service info
    """

    def __init__(self, data: str) -> None:
        info = data.split('\t')

        self.container_id = info[0].split('@')[-1]
        self.name = info[1].split('@')[-1]
        self.status = info[2].split('@')[-1]
        self.ports = info[3].split('@')[-1]
        self.created_at = info[4].split('@')[-1]
    
    def serialize(self) -> dict:
        return {
            "container_id": self.container_id,
            "name": self.name,
            "status": self.status,
            "ports": self.ports,
            "created_at": self.created_at
        }